packages/cloudflare/src/api/get-cloudflare-context.ts

```diff
@@ -0,0 +1,50 @@
+import "server-only";
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-empty-interface
+  interface CloudflareEnv {}
+}
+
+export type CloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+> = {
+  /**
+   * object containing the worker's bindings ([docs](https://developers.cloudflare.com/workers/runtime-apis/bindings/))
+   */
+  env: CloudflareEnv;
+  /**
+   * object containing information regarding the current request ([docs](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties))
+   */
+  cf: CfProperties;
+  /**
+   * the current execution context object ([docs](https://developers.cloudflare.com/workers/runtime-apis/context))
+   */
+  ctx: Context;
+};
+
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
```

Comment:
Add a note that this throws when the context cannot be retrieved?

Reply:
It should not really throw in the final implementation tough...

I don't mind adding the comment for now